src/font.c

```diff
@@ -50,6 +104,46 @@ void font_create_ctfont(struct font* font) {
   CTFontDescriptorRef descriptor = CTFontDescriptorCreateWithAttributes(attr);
 
   if (font->ct_font) CFRelease(font->ct_font);
+
+  if (font->features) {
+    char* features_copy = string_copy(font->features);
+    char* feature = strtok(features_copy, ",");
+
+    while (feature) {
+      int feature_name = 0;
+      int feature_selector = 0;
+
+      bool valid_feature = false;
+
+      if (sscanf(feature, "%d:%d", &feature_name, &feature_selector) == 2) {
+        valid_feature = true;
+      } else if (strlen(feature) == 4) {
+        get_truetype_feature(feature, &feature_name, &feature_selector);
+
+        if (feature_name != 0) {
+          valid_feature = true;
+        }
+      }
+
+      if (valid_feature) {
+        CFNumberRef name = CFNumberCreate(NULL, kCFNumberIntType, &feature_name);
+        CFNumberRef value = CFNumberCreate(NULL, kCFNumberIntType, &feature_selector);
+
+        CTFontDescriptorRef new_descriptor = CTFontDescriptorCreateCopyWithFeature(descriptor, name, value);
+        CFRelease(descriptor);
+        descriptor = new_descriptor;
+
+        CFRelease(name);
+        CFRelease(value);
+      }
+
+      feature = strtok(NULL, ",");
+    }
+
+    free(feature);
```

Comment:
I think we should not free the pointer returned from strtok, as it could result in a double free